data/transactions/logic/eval.go

```diff
@@ -205,39 +364,50 @@ func Eval(program []byte, params EvalParams) (pass bool, err error) {
 		cx.err = fmt.Errorf("program version %d greater than max supported version %d", version, EvalMaxVersion)
 		return false, cx.err
 	}
-	if version > params.Proto.LogicSigVersion {
-		cx.err = fmt.Errorf("program version %d greater than protocol supported version %d", version, params.Proto.LogicSigVersion)
+	if version > cx.EvalParams.Proto.LogicSigVersion {
+		cx.err = fmt.Errorf("program version %d greater than protocol supported version %d", version, cx.EvalParams.Proto.LogicSigVersion)
 		return false, cx.err
 	}
-	if (params.Txn.Txn.RekeyTo != basics.Address{}) { // Currently no TEAL version knows about the RekeyTo field, but this may change in a future TEAL version
+	if cx.EvalParams.Txn.Txn.RekeyTo != (basics.Address{}) { // Currently no TEAL version knows about the RekeyTo field, but this may change in a future TEAL version
```

Comment:
I think that using !cx.EvalParams.Txn.Txn.RekeyTo.Zero() would be cleaner

Reply:
done